packages/create-cloudflare/src/helpers/poll.ts

```diff
@@ -0,0 +1,74 @@
+import { Resolver } from "node:dns/promises";
+import { request } from "undici";
+import { blue, brandColor, dim } from "./colors";
+import { spinner } from "./interactive";
+
+const TIMEOUT = 1000 * 60 * 5;
+const POLL_INTERVAL = 1000;
+
+export const poll = async (url: string): Promise<boolean> => {
+	const start = Date.now();
+	const domain = new URL(url).host;
+	const s = spinner();
+
+	s.start("Waiting for DNS to propagate");
+	while (Date.now() - start < TIMEOUT) {
+		s.update(`Waiting for DNS to propagate (${secondsSince(start)}s)`);
+		if (await dnsLookup(domain)) {
+			s.stop(`${brandColor("DNS propagation")} ${dim("complete")}.`);
+			break;
+		}
+		await sleep(POLL_INTERVAL);
+	}
+
+	s.start("Waiting for deployment to become available");
```

Comment:
looks like you're calling start again on the same spinner which looks like it might leave 2 setIntervals going?

Reply:
I think this is ok actually. I don't think this line will get encountered until the dns lookup completes, at which point it will have called `s.stop` and cleared the interval.

Reply:
Except if we timeout... but then we should bail early and not start the next spinner?

Reply:
Resolved by adding interval clear to start.